### PR TITLE
Changes to use subject as user when different to client_id

### PR DESCRIPTION
**Pull request by dnyanesh-sonwane**
This PR fixes bug of broken VPD due to incorrect usage of client credentials
https://trello.com/c/UBTQQpL4/727-%F0%9F%90%9B-bug-delius-api-vpd-broken-for-client-credentials

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/config/AuthAwareTokenConverter.kt [hunk 1]
```diff
@@ -19,10 +19,15 @@ class AuthAwareTokenConverter(
...
-        jwt.getClaimAsString(ClaimNames.DATABASE_USERNAME)
-          ?: throw BadRequestException("Database username required for client credentials")
-      )
+      "none" -> {
+        val username = if (!jwt.subject.equals(jwt.claims["client_id"])) {
+          jwt.subject
+        } else {
+          jwt.getClaimAsString(ClaimNames.DATABASE_USERNAME)
+            ?: throw BadRequestException("Database username required for client credentials")
+        }
```
Comment (axle-h) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/config/AuthAwareTokenConverter.kt:28]: I think an explainer comment here would make sense

Comment (dnyanesh-sonwane) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/config/AuthAwareTokenConverter.kt:30]: done

**Review by axle-h — APPROVED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/deliusapi/integration/JwtAuthHelper.kt [hunk 2]
```diff
@@ -29,6 +30,7 @@ class JwtAuthHelper {
   fun createJwt(
     subject: String?,
     userId: Long? = Fake.id(),
+    clientId: String? = "elite2apiclient",
```
Comment (mxco86) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/deliusapi/integration/JwtAuthHelper.kt:33]: Nitpick: could we remove the reference to 'elite2'?

**Review by mxco86 — APPROVED** [review]
Tiny comment

File: src/test/kotlin/uk/gov/justice/digital/hmpps/deliusapi/integration/JwtAuthHelper.kt (hunk 2 above)
Comment (dnyanesh-sonwane) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/deliusapi/integration/JwtAuthHelper.kt]: done

**Pull request merged**